boxsdk/util/log.py

```diff
@@ -1,12 +1,42 @@
 # coding: utf-8
 
 from __future__ import unicode_literals
+
 import logging
+try:
+    from logging import NullHandler
+except ImportError:
+    class NullHandler(logging.Handler):
+        def emit(self, record):
+            pass
 import sys
 
 from six import string_types
 
 
+class Logging(object):
+    _has_setup = False
+
+    def setup_logging(self, stream_or_file=None, debug=False, name=None):
+        if not self._has_setup:
+            self._has_setup = True
+            return self._setup_logging(stream_or_file, debug, name)
```

Comment:
Why do we need to return a logger? Can't this be a method that is used 100% for side-effects, and have clients always use the standard `logger.getLogger` to get the logger?

The fact that it can also return `None` means that you can't really rely on the return value anyway.

It looks like all the code in this PR is already doing that, we never access the return value except in tests.

It looks like removing the return value would be a breaking change, but I think that's fine. It's easy enough to deal with the change, and I doubt many people were using this function anyway.